jte/src/main/java/gg/jte/resolve/CompositeCodeResolver.java

```diff
@@ -0,0 +1,68 @@
+package gg.jte.resolve;
+
+
+import gg.jte.CodeResolver;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Resolves template code from multiple other resolvers
+ */
+public class CompositeCodeResolver implements CodeResolver {
+
+    private final List<CodeResolver> codeResolvers;
+
+    public CompositeCodeResolver(List<CodeResolver> codeResolvers) {
+        this.codeResolvers = codeResolvers;
+    }
+
+    @Override
+    public String resolve(String name) {
+        var content = "";
+        for (CodeResolver codeResolver : this.codeResolvers) {
+            try {
+                String resolve = codeResolver.resolve(name);
+                if(Objects.nonNull(resolve)){
+                    content = resolve;
+                }
+            } catch (Exception ex) {
```

Comment:
`CodeResolver` does not declare any checked exceptions. I don't think it is appropriate to be swallowing downstream exceptions here. Are you encountering exceptions in your usage?

Reply:
The following implementations of DirectoryCodeResolver and ResourceCodeResolver each throw exceptions.

DirectoryCodeResolver 

```java
    @Override
    public String resolve(String name) {
        try {
            Path file = root.resolve(name);
            return Files.readString(file, StandardCharsets.UTF_8);
        } catch (NoSuchFileException e) {
            return null;
        } catch (IOException e) {
            throw new UncheckedIOException(e);
        }
    }
```

ResourceCodeResolver

```java
 @Override
    public String resolve(String name) {
        try (InputStream is = getClassLoader().getResourceAsStream(root + name)) {
            if (is == null) {
                return null;
            }
            return IoUtils.toString(is);
        } catch (IOException e) {
            throw new UncheckedIOException(e);
        }
    }
```

But I don't have to throw an exception in the loop's catch clause. I propose this new implementation below

```java
    @Override
    public String resolve(String name) {
        var content = "";
        for (CodeResolver codeResolver : this.codeResolvers) {
            try {
                String resolve = codeResolver.resolve(name);
                if(resolve != null){
                    content = resolve;
                }
            } catch (Exception e) {
                // ignore
            }
        }
        if(!content.isEmpty())
            return content;
        return null;
    }
```

Reply:
As @edward3h suggested, it is not a good idea to swallow exceptions like this. Both code resolvers return `null` if the template does not exist. In case I'm loading a template that exists, but there's an IO error during loading, users want to know this happened.